### PR TITLE
Initial support for newer packet versions

### PR DESCRIPTION
**Pull request by Lemongrass3110**
* **Addressed Issue(s)**: #3925, #4419

* **Server Mode**: Both

* **Description of Pull Request**: 
Basic support for packets up to April 2020
Changing the default packet version to 2020-04-01

Special thanks and credits to @4144 for keeping the basic packet data updated and maintaining it
as well as to @shennetsind for the initial design idea.

Not all features are supported yet - please refrain from asking for them in here.
Some clients < 2010 are still broken and will be fixed during the code review / testing time.

### CODE REVIEW

File: src/map/mob.cpp [hunk 1]
```diff
@@ -5563,7 +5563,7 @@ static void mob_load(void)
 
 		sv_readdb(dbsubpath1, "mob_chat_db.txt", '#', 3, 3, -1, &mob_parse_row_chatdb, silent);
 
-		if( db_use_sqldbs && i == 0 )
+		if( db_use_sqldbs )
```
Comment (RagnaWay) [inline src/map/mob.cpp:5566]: Need this `if( db_use_sqldbs && i == 0 )`

`[Error]: mob_parse_row_mobskilldb: Too many skills for monster 1150[MOONLIGHT]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 1511[AMON_RA]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 1637[MAGALETA]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 1639[KATRINN]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 1643[G_MAGALETA]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 1646[B_SEYREN]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 1647[B_EREMES]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 1649[B_MAGALETA]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 1650[B_SHECIL]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 1651[B_KATRINN]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 1719[DETALE]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 1734[KIEL_]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 1751[RANDGRIS]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 1779[KTULLANUX]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 1870[NECROMANCER]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 1871[FALLINGBISHOP]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 1874[BEELZEBUB_]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 1916[MOROCC]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 1917[MOROCC_]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 1929[BAPHOMET_I]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 1947[PIAMETTE_]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 1956[NAGHT_SIEGER]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 1990[HARDROCK_MOMMOTH]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2009[WOOMAWANG_]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2022[S_NYDHOG]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2103[E_KTULLANUX]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2107[E_FALLINGBISHOP]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2109[E_RANDGRIS]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2112[E_B_SEYREN]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2113[E_B_EREMES]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2131[LOST_DRAGON]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2156[LEAK]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2189[COELACANTH_H_M]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2190[COELACANTH_H_A]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2194[MD_GIANT_OCTOPUS]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2202[KRAKEN]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2223[CELIA]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2228[G_RANDEL]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2230[G_CELIA]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2232[G_GERTIE]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2235[B_RANDEL]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2236[B_FLAMEL]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2237[B_CELIA]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2238[B_CHEN]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2239[B_GERTIE]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2240[B_ALPHOCCIO]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2241[B_TRENTINI]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2362[N_AMON_RA]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2419[L_KATRINN]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2432[G_L_KATRINN]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2736[C3_NECROMANCER]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2875[C3_CELIA]
[Error]: mob_parse_row_mobskilldb: Too many skills for monster 2999[EP14_MORS_BOSSA]`

**Comment by arthurcunhamello** [comment]
Error Hexed 2020 03 18

[Info]: 'Testeee' logged in. (AID/CID: '2000001/150002', IP: '127.0.0.1', Group '99').
[Warning]: clif_parse: Received unsupported packet (packet 0x0b6d, 6 bytes received), disconnecting session #3.
[Info]: Character 'Testeee' logged off.

when clicking on the shop

**Comment by RagnaWay** [comment]
> Error Hexed 2020 03 18
> 
> [Info]: 'Testeee' logged in. (AID/CID: '2000001/150002', IP: '127.0.0.1', Group '99').
> [Warning]: clif_parse: Received unsupported packet (packet 0x0b6d, 6 bytes received), disconnecting session #3.
> [Info]: Character 'Testeee' logged off.
> 
> when clicking on the shop
> 
> the hexed 2020 04 01 I can't use the keyboard.

I have the same error. Testing in **2020-02-06**
`[Warning]: clif_parse: Received unsupported packet (packet 0x0b6d, 6 bytes received), disconnecting session #15.`

**Comment by alisonrag** [comment]
> > Error Hexed 2020 03 18
> > 
> > [Info]: 'Testeee' logged in. (AID/CID: '2000001/150002', IP: '127.0.0.1', Group '99').
> > [Warning]: clif_parse: Received unsupported packet (packet 0x0b6d, 6 bytes received), disconnecting session #3.
> > [Info]: Character 'Testeee' logged off.
> > 
> > when clicking on the shop
> > 
> > the hexed 2020 04 01 I can't use the keyboard.
> 
> I have the same error. Testing in **2020-02-06**
> `[Warning]: clif_parse: Received unsupported packet (packet 0x0b6d, 6 bytes received), disconnecting session #15.`

To enable keyboard packet 0B1B must be sended from server to client after hotkey list

https://github.com/OpenKore/openkore/pull/3074

**Comment by RagnaWay** [comment]
It is crashing when clicking on the (Adventurer's agent) icon.
Is there a way to add a restriction to avoid the crash?
image

**Comment by Balferian** [comment]
> It is crashing when clicking on the (Adventurer's agent) icon.
> Is there a way to add a restriction to avoid the crash?
> image

I test it on 2020-04-01 exe + last kRo client and don`t see server/client crash.

**Comment by RagnaWay** [comment]
> I test it on 2020-04-01 exe + last kRo client and don`t see server/client crash.

Don't give an error to you when clicking the button?

**Comment by Balferian** [comment]
> > I test it on 2020-04-01 exe + last kRo client and don`t see server/client crash.
> 
> Don't give an error to you when clicking the button?

I didn`t update my msgstring yet, but window opens
image

**Comment by RagnaWay** [comment]
> > > I test it on 2020-04-01 exe + last kRo client and don`t see server/client crash.
> > 
> > 
> > Don't give an error to you when clicking the button?
> 
> I didn`t update my msgstring yet, but window open
> image

Thanks

**Comment by 4144** [comment]
Please add hercules copyrights in header of files where you copy code from hercules.

Because here reused too much hercules code, and no any references to hercules.

**Comment by ecdarreola** [comment]
I'm unable to use skill or use macro in-game like "alt+s" and "alt+m", am I missing something with the PR? or is it client related files?

**Comment by Akkarinage** [comment]
> Please add hercules copyrights in header of files where you copy code from hercules.
> 
> Because here reused too much hercules code, and no any references to hercules.

You again, with your usual "give us credits" - practice what you preach, sir, or don't bother preaching at all.

We do not come sprawling all over your commits demanding credit, because we honestly don't care when you don't give rAthena or it's devs credit for work they've done, the rA devs are bigger than that.

I, on the other hand, am sick of your moaning. Not satisfied with being credited personally in the PR (as it's you that has done the work, from my understanding) you're moaning about your project's name not being in the code? Do you put rAthena in your headers? No, the Hercules project was very quick to remove all traces of rAthena within it's code after the pointless fork.

Hercules devs don't correctly attribute credits, so I've urged the rAthena devs to ignore your drama and continue giving credits where credit is due.

**Comment by alisonrag** [comment]
> I'm unable to use skill or use macro in-game like "alt+s" and "alt+m", am I missing something with the PR? or is it client related files?

Missing 0B1B

Check my comment:
https://github.com/rathena/rathena/pull/4944#issuecomment-628115489

@Lemongrass3110 can u check this?

**Comment by Balferian** [comment]
1. After change job skills don`t remove from skill bar (need relogin).
2. If player use refresh, impossible to use hot keys F1-F9 / ALT+...
(looks like client want some packet or problem in refresh command in new client)

**Comment by 4144** [comment]
> > Please add hercules copyrights in header of files where you copy code from hercules.
> > Because here reused too much hercules code, and no any references to hercules.
> 
> You again, with your usual "give us credits" - practice what you preach, sir, or don't bother preaching at all.
> 
> We do not come sprawling all over your commits demanding credit, because we honestly don't care when you don't give rAthena or it's devs credit for work they've done, the rA devs are bigger than that.
> 
> I, on the other hand, am sick of your moaning. Not satisfied with being credited personally in the PR (as it's you that has done the work, from my understanding) you're moaning about your project's name not being in the code? Do you put rAthena in your headers? No, the Hercules project was very quick to remove all traces of rAthena within it's code after the pointless fork.
> 
> Hercules devs don't correctly attribute credits, so I've urged the rAthena devs to ignore your drama and continue giving credits where credit is due.

I never copy rathena code without credits. because code was too small i not add copyrights in code, but only credits in commit messages.

Also hercules already include rathena copyrights. All devs and teams present in authors.
In rathena hercules devs present in authors, but no any references to eathena/athena or other teams in headers.

Hercules is GPL code, but who copy code must follow also copyrights. Code always have some authors.

And actually @Akkarinage you was not asked about copyrights. I know some ra devs including you hate all hercules devs. May be this is past wars, idk.

My message was to @Lemongrass3110.

**Comment by Akkarinage** [comment]
> And actually @Akkarinage you was not asked about copyrights. I know some ra devs including you hate all hercules devs. May be this is past wars, idk.

This statement right here is why we can't get along. No rAthena dev hates a Herc dev.
Grow up. This pettiness is what's keeping the divide alive.

**Comment by 4144** [comment]
> > And actually @Akkarinage you was not asked about copyrights. I know some ra devs including you hate all hercules devs. May be this is past wars, idk.
> 
> This statement right here is why we can't get along. No rAthena dev hates a Herc dev.
> Grow up. This pettiness is what's keeping the divide alive.

Your attacks on me only confirming my words...

**Comment by Akkarinage** [comment]
Garbage has been removed to the forums.

**Comment by RagnaWay** [comment]
> Client 2020-04-01
> 
> 1. After change job skills don`t remove from skill bar (need relogin).
> 2. If player use refresh, impossible to use hot keys F1-F9 / ALT+...
>    (looks like client want some packet or problem in refresh command in new client)
> 3. Attandance window doesn`t open.
> 4. If change config in game "데미지 표기 확장" to show damage digits are croped.
>    image
Attandance working, however it is not showing the remaining day, and the event period correctly.
image

**Comment by Balferian** [comment]
> >    Attandance working, however it is not showing the remaining day, and the event period correctly.
> >    image

Thanks, I`ll check my configs/luas.

**Comment by RagnaWay** [comment]
> Client 2020-04-01
> 
> 1. After change job skills don`t remove from skill bar (need relogin).
> 2. If player use refresh, impossible to use hot keys F1-F9 / ALT+...
>    (looks like client want some packet or problem in refresh command in new client)
> 3. ~Attandance window doesn`t open.~
> 4. If change config in game "데미지 표기 확장" to show damage digits are croped.
>    image
> 5. Guild icons don`t work.
>    image
> 6. Items don`t add from cash shop to inventory when deal is confirmed.
>    ([Warning]: Player 150000 sent incorrect cash shop buy packet (len 20:24)!)
> 7. Need to update Madogear packets to send params: 0 - old mado / 1 - new mado, for now mado don`t show on character.
> 8. When player try to take items from mail client crashed (only with this button).
>    After relogin items are in inventory.
>    image

**Mail (Rodex)**: Use this msgstringtable: https://github.com/llchrisll/ROenglishRE/blob/master/Ragnarok/data/msgstringtable.txt
**Guild Emblem**: 
From the client date **2019-06-05** the guild emblem uses the uploaded/downloaded though an HTTP connection Used in lua files/service_korea/

**Comment by secretdataz** [comment]
Guild icons requires an external web service to provide upload and download API to function.
Yesterday I've spun up an experimental project for this use case. Please help me test it. Thanks :)

https://github.com/secretdataz/athena-web-service

cc @Balferian

**Comment by RagnaWay** [comment]
> Guild icons requires an external web service to provide upload and download API to function.
> Yesterday I've spun up an experimental project for this use case. Please help me test it. Thanks :)
> 
> https://github.com/secretdataz/athena-web-service
> 
> cc @Balferian

I will test it now. <3

**Comment by alisonrag** [comment]
> Guild icons requires an external web service to provide upload and download API to function.
> Yesterday I've spun up an experimental project for this use case. Please help me test it. Thanks :)
> 
> https://github.com/secretdataz/athena-web-service
> 
> edit: yes, there's no documentation, for now
> 
> cc @Balferian

this webservice also provide support to User Config? or only emblem?

**Comment by LolyAll** [comment]
which client version is best to test? I have 20191224 but every time I type login and password the game crashes, nor does it even arrive at the emulator.

**Comment by JohnnyPlayy** [comment]
> which client version is best to test? I have 20191224 but every time I type login and password the game crashes, nor does it even arrive at the emulator.

Your problem must be related to the client or data

**Comment by LolyAll** [comment]
@JohnnyPlayy have a specific patch to connect? I managed to fix the crash problem, but when I type in the login data it doesn't connect, do I have to turn off packet encrypt? have to do something to be able to connect?

**Comment by JohnnyPlayy** [comment]
> @JohnnyPlayy have a specific patch to connect? I managed to fix the crash problem, but when I type in the login data it doesn't connect, do I have to turn off packet encrypt? have to do something to be able to connect?

Checks if your problem is not ClientInfo
image
This is how the ClientInfo configuration should be

Anything calls me on discord

**Comment by attackjom** [comment]
these skill need to change a bit to use properly new skilleffect.

1. Gravitational Field  -> change unit ID from Gravitation -> Dummyskill (If not the skill will make too much lag and sometimes game will crash)

2. Earth Drive -> change from clif_skill_damage -> clif_skill_nodamage  to enable beautiful ground crack effect.

**Comment by Questune09** [comment]
Just 2 reading login and mapserver how to solve this?
image

**Comment by admkakaroto** [comment]
Client 20200401, my test:

- Mado is not being shown, if try remove received: [Warning]: clif_parse: Received unsupported packet (packet 0x0b35, 3 bytes received), disconnecting session #4.

- If use @refresh, are not able to use any shortcut or hot key.

- Trying to put items in the cart, is not displaying the items, it is necessary to use refresh or relog to appear.

```
Alert trying to compile:
\rathena\src\map\clif.cpp(3899,18): warning C4146: unary minus operator applied to unsigned type, result still unsigned
\rathena\src\map\clif.cpp(3901,18): warning C4146: unary minus operator applied to unsigned type, result still unsigned
\rathena\src\map\clif.cpp(3917,18): warning C4146: unary minus operator applied to unsigned type, result still unsigned
\rathena\src\map\clif.cpp(3919,18): warning C4146: unary minus operator applied to unsigned type, result still unsigned
```

- CashShop not open, causing crash all time.

I also got the message 'clif_name bad name (64)', but I still don't know how to reproduce it, I'll try to find out.

**Comment by RagnaWay** [comment]
> Client 20200401, my test:
> 
> * Mado is not being shown, if try remove received: [Warning]: clif_parse: Received unsupported packet (packet 0x0b35, 3 bytes received), disconnecting session #4.
> * If use @refresh, are not able to use any shortcut or hot key.
> * Trying to put items in the cart, is not displaying the items, it is necessary to use refresh or relog to appear.
> 
> ```
> Alert trying to compile:
> \rathena\src\map\clif.cpp(3899,18): warning C4146: unary minus operator applied to unsigned type, result still unsigned
> \rathena\src\map\clif.cpp(3901,18): warning C4146: unary minus operator applied to unsigned type, result still unsigned
> \rathena\src\map\clif.cpp(3917,18): warning C4146: unary minus operator applied to unsigned type, result still unsigned
> \rathena\src\map\clif.cpp(3919,18): warning C4146: unary minus operator applied to unsigned type, result still unsigned
> ```
> 
> * CashShop not open, causing crash all time.
> 
> I also got the message 'clif_name bad name (64)', but I still don't know how to reproduce it, I'll try to find out.
> 
> Some things I think have already been reported
The **cashshop** is working perfectly with me, Lemongrass has already sent the correction, look at the history of PR.

**Comment by voyfmyuh** [comment]
Client 20200401, my test:
註解 2020-05-18 042930

How to solve database user_pass record as token?

**Comment by alisonrag** [comment]
> Client 20200401, my test:
> 註解 2020-05-18 042930
> 
> How to solve database user_pass record as token?

2020 clients use a weblauncher that use token to login instead of password
Maybe sometime someone will release it.
Can u use restore default login screen in Nemo?

**Comment by voyfmyuh** [comment]
Client 20200401, my test:
When langtype is set to 4, user_pass will be recorded as garbled
註解 2020-05-18 062132

**Comment by ecdarreola** [comment]
> **shshop** is working perfectly with me, Lemongrass has already sent the correction, look at the history of PR.

What client version are you using? 

For Cash Shop:
Using 2020-04-01: 
image

**Comment by RagnaWay** [comment]
> > **shshop** is working perfectly with me, Lemongrass has already sent the correction, look at the history of PR.
> 
> What client version are you using?
> 
> For Cash Shop:
> Using 2020-04-01:
> image

same 2020-04-01 I've tested everything at the cashshop and it's okay.

**Comment by ecdarreola** [comment]
> > > **shshop** is working perfectly with me, Lemongrass has already sent the correction, look at the history of PR.
> > 
> > 
> > What client version are you using?
> > For Cash Shop:
> > Using 2020-04-01:
> > image
> 
> same 2020-04-01 I've tested everything at the cashshop and it's okay.

This is weird. I've tested a new one using fresh install and just the PR i'm getting the same result of incorrect cash shop buy packet at 2020-04-01.

**Comment by voyfmyuh** [comment]
Client 20200401
Click the exit window to disconnect directly
註解 2020-05-19 072028

**Comment by LolyAll** [comment]
Any news or fixes?

**Comment by Surefirer** [comment]
Does this require a special client to run? I am using the lastest KRO client, and it gave me [B5031], I did not find any guide to set up the client-side. Sorry if this is the wrong place, but I appreciate if someone can guide me. Thanks.

**Comment by aleos89** [comment]
By "latest", I assume you mean up to April 1, 2020 like the original post states.

**Comment by Surefirer** [comment]
> By "latest", I assume you mean up to April 1, 2020 like the original post states.

yes, please see my set up below.

**Comment by Balferian** [comment]
@qwerty7vp Yeah, I already solved the problem, a had old digit sprite.

**Comment by qwerty7vp** [comment]
I have found Storage problem

- I want to pick This item into Inventory
image

- It give me below item but still have That item icon there
image

- Then I closed a Storage and re-open it again so I found the item I want to pick at first place
image

- Now I want to Store item that I just got from Storage back
image

- It ended up became other item icon
image

Tested by 20200304 and 20200401 clients

**Comment by llchrisll** [comment]
I also found an bug with Dragon mount for RK:
```[Warning]: clif_parse: Received unsupported packet (packet 0x0b35, 3 bytes received), disconnecting session #3.```
Mounting itself works fine, but unmounting makes you disconnect and shows the error mentioned.

**Comment by admkakaroto** [comment]
> I also found an bug with Dragon mount for RK:
> `[Warning]: clif_parse: Received unsupported packet (packet 0x0b35, 3 bytes received), disconnecting session #3.`
> Mounting itself works fine, but unmounting makes you disconnect and shows the error mentioned.
> 
> Client: 2020-04-01 Ragexe

Yes, I have commented on this, any kind of mount, cart, peco, dragon, mado or anything else.

**Comment by Lemongrass3110** [comment]
> 
> 
> these skill need to change a bit to use properly new skilleffect.
> 
>     1. Gravitational Field  -> change unit ID from Gravitation -> Dummyskill (If not the skill will make too much lag and sometimes game will crash)
> 
>     2. Earth Drive -> change from clif_skill_damage -> clif_skill_nodamage  to enable beautiful ground crack effect.

Do you know when these were exactly changed?

**Comment by Lemongrass3110** [comment]
> 
> 
> > I'm unable to use skill or use macro in-game like "alt+s" and "alt+m", am I missing something with the PR? or is it client related files?
> 
> Missing 0B1B
> 
> Check my comment:
> [#4944 (comment)](https://github.com/rathena/rathena/pull/4944#issuecomment-628115489)
> 
> @Lemongrass3110 can u check this?

@alisonrag this should have only been an issue if you use `@refresh`. Otherwise the client should send CZ_BLOCKING_PLAY_CANCEL after it finished loading and sending 0xB1B will be triggered.

**Comment by RagnaWay** [comment]
When using this diff, RefineUI and Extended Vendings no longer work.
image
image
I do not know if it is an error with this diff of packets 2020 or if you need to update RefineUI and Extended Vendings to work on this packet.

**Comment by admkakaroto** [comment]
@RagnaWay these custons diffs need to be updated

**Comment by qwerty7vp** [comment]
I tested the mado is showing now
anyone know how to change between oldmado and newmado ?
it is atcommand like bodystyle or what

**Comment by Lemongrass3110** [comment]
> 
> 
> I tested the mado is showing now
> anyone know how to change between oldmado and newmado ?
> it is atcommand like bodystyle or what

Not supported right now. Features will be added later on, when this is on master.

**Comment by Lemongrass3110** [comment]
> 
> 
> @RagnaWay these custons diffs need to be updated

@admkakaroto Correct!

**Comment by qwerty7vp** [comment]
I found problem about `show_mob_info:`

normally when use /showname
type 1 should show Mob's name with bold font
type 2 should show Mob's name and mob's info with thin font

but this is what I've got from `show_mob_info: 6`

with showname type 1

image


with showname type 2

image

then I tried with `show_mob_info: 1`

image

Client : 20200304

**Comment by Lemongrass3110** [comment]
> 
> 
> I found problem about `show_mob_info:`
> 
> normally when use /showname
> type 1 should show Mob's name with bold font
> type 2 should show Mob's name and mob's info with thin font
> 
> but this is what I've got from `show_mob_info: 6`
> 
> with showname type 1
> 
> image
> 
> with showname type 2
> 
> image
> 
> then I tried with `show_mob_info: 1`
> 
> image
> 
> Client : 20200304

@qwerty7vp 
Could you tell us what your exact problem is?
I do not see the problem...

**Comment by qwerty7vp** [comment]
I mean
Normally it should be like this right ?

with showname type 1
image

with showname type 2
image

both pictures using `show_mob_info: 6`
Client : 20180919 without this PR

**Comment by Angelic234** [comment]
Can someone test => storage items more then 595 or 597? I got map-server when I open storage.

**Comment by qwerty7vp** [comment]
> Can someone test => storage items more then 595 or 597? I got map-server when I open storage.

Confirmed this

Map-server start to crash at 101/600 items in storage

**Comment by admkakaroto** [comment]
A new problem appeared on the cart, I can't remove the last item from the list, the items sent to the cart are still not being updated correctly.

a1111

You also cannot remove all items from the cart, there will always be 1 item.

**Comment by qwerty7vp** [comment]
> A new problem appeared on the cart, I can't remove the last item from the list, the items sent to the cart are still not being updated correctly.
> 
> a1111
> 
> You also cannot remove all items from the cart, there will always be 1 item.

I guess the problem happen when you have items in the cart from old server data (before pull this PR)

because I had this problem too
but when I tried to remove item from cart from server **without this PR + old client**
and tried to use cart again **with this PR + new client** again
there's no problem anymore

**Comment by admkakaroto** [comment]
Guild storage log, are not working too.

**Comment by LolyAll** [comment]
receiving crash on mapserver if have more 100 items in storage too.

**Comment by Lemongrass3110** [comment]
In case someone needs it to test the storages:
```
insert into storage(account_id,nameid,amount)
select 2000000,id,1 from item_db_re order by id asc limit 1,900
```

**Comment by Lemongrass3110** [comment]
Anything left that I missed?

**Comment by qwerty7vp** [comment]
@Lemongrass3110 so the `show_mob_info:` does not bugged?
since it's not working the same like I sent 2 of my comments above
or it's up to client version ?

**Comment by LolyAll** [comment]
guild emblem still not working

**Comment by Lemongrass3110** [comment]
> guild emblem still not working

@LolyAll read @secretdataz reply above. Not part of this pull request.

**Comment by ecdarreola** [comment]
Not sure if this will be relevant but an error appeared after being afk on a local server. Don't know how to reproduce it.

image

**Comment by Balferian** [comment]
> Not sure if this will be relevant but an error appeared after being afk on a local server. Don't know how to reproduce it.
> 
> image

I have this error when cast aoe (ШтормWZ_STORMGUST)  skills on mobs.

**Comment by Balferian** [comment]
After player buy item from marketshop in chat showed message "Not enough zeny", but zeny is enough and item added to inventory.
Can anyone confirm it?

**Comment by qwerty7vp** [comment]
> I have this error when cast aoe skills (WZ_STORMGUST) on mobs.
confirmed this
skills that I tested and found error ( WZ_STORMGUST, WZ_VERMILION, WZ_HEAVENDRIVE, HW_GRAVITATION )
maybe there are more but not tested yet
image

**Comment by alisonrag** [comment]
> After player buy item from marketshop in chat showed message "Not enough zeny", but zeny is enough and item added to inventory.
> Can anyone confirm it?

MARKET shop use 2 differents responses depending on clientversion, my guess is that rAthena dont updated the  response to new clients

**Comment by Balferian** [comment]
Client crashed when use skills TalkBox (hunter) and Graffiti (rogue) after input text.

**Comment by admkakaroto** [comment]
If try to use 'Talkie Box', receiving:

```
[Warning]: clif_parse: Received unsupported packet (packet 0xe085, 2 bytes received), disconnecting session #3.
[Warning]: clif_parse: Received unsupported packet (packet 0xc527, 3 bytes received), disconnecting session #3.
```

**Comment by Angelic234** [comment]
Can someone test =>
1. storage items more then 599
2. close storage
3. logout and login
4. map crash

**Comment by qwerty7vp** [comment]
> Can someone test =>
> 
> 1. storage items more then 599
> 2. close storage
> 3. logout and login
> 4. map crash

fixed in 05807b6
pull the updated commits

**Comment by Angelic234** [comment]
> fixed in [05807b6](https://github.com/rathena/rathena/commit/05807b6d98ab00a371f2cb5839880b6593da2dc6)
> pull the updated commits
I test this pull , try my step, when u relogin map crash

**Comment by qwerty7vp** [comment]
> I test this pull , try my step, when u relogin map crash
 >

gweg

**Comment by Angelic234** [comment]
ok~ thanks ， I'ii try find my program.

**Comment by qwerty7vp** [comment]
I found a bug that when Store item in Storage more than 102
and Equipping Items start to bugged
and take item out after bugged not helping

https://youtu.be/TOGNkTDMxZ8

**Comment by admkakaroto** [comment]
Sales still not working, 
there is a strange bug in the equipment window, if you are BG several times, at some point it bug, you click to equip the item but it is not displayed in the equipment or costume window, but if you use refresh the items are displayed, I will try to reproduce that.

**Comment by Lemongrass3110** [comment]
> 
> 
> Sales still not working,
> there is a strange bug in the equipment window, if you are BG several times, at some point it bug, you click to equip the item but it is not displayed in the equipment or costume window, but if you use refresh the items are displayed, I will try to reproduce that.

@admkakaroto sales have new packets and cannot support them for now.

**Comment by Feelmeone** [comment]
Equipment is not displayed when viewing the equipment window of another character.
image

**Comment by admkakaroto** [comment]
> Equipment is not displayed when viewing the equipment window of another character.
> image

I confirm, client 20200401

File: src/char/char.cpp [hunk 2]
```diff
@@ -924,7 +924,8 @@ int char_mmo_chars_fromsql(struct char_session_data* sd, uint8* buf, uint8* coun
 		"`str`,`agi`,`vit`,`int`,`dex`,`luk`,`max_hp`,`hp`,`max_sp`,`sp`,"
 		"`status_point`,`skill_point`,`option`,`karma`,`manner`,`hair`,`hair_color`,"
 		"`clothes_color`,`body`,`weapon`,`shield`,`head_top`,`head_mid`,`head_bottom`,`last_map`,`rename`,`delete_date`,"
-		"`robe`,`moves`,`unban_time`,`font`,`uniqueitem_counter`,`sex`,`hotkey_rowshift`,`title_id`,`show_equip`"
+		"`robe`,`moves`,`unban_time`,`font`,`uniqueitem_counter`,`sex`,`hotkey_rowshift`,`title_id`,`show_equip`,"
+		"`hotkey_rowshift`"
```
Comment (Atemo) [inline src/char/char.cpp:928]:
```suggestion
		"`hotkey_rowshift2`"
```

**Comment by LolyAll** [comment]
There is another bug, when you put items in the merchant's cart, not all items appear, you need to use @refresh to appear, some appear, I will record a video to show how the bug is.

**Comment by qwerty7vp** [comment]
> There is another bug, when you put items in the merchant's cart, not all items appear, you need to use @refresh to appear, some appear, I will record a video to show how the bug is.

I comfirm this

ezgif-1-db363f1ae3d7

some items disappear when put them in cart
and came back after `@refresh`

File: src/map/clif.cpp [hunk 3]
```diff
@@ -17350,7 +17233,7 @@ void clif_mercenary_info(struct map_session_data *sd)
 
 
 /// Mercenary skill tree (ZC_MER_SKILLINFO_LIST).
-/// 029d <packet len>.W { <skill id>.W <type>.L <level>.W <sp cost>.W <attack range>.W <skill name>.24B <upgradable>.B }*
+/// 029d <packet len>.W { <skill id>.W <type>.L <level>.W <sp cost>.W <attack range>.W <skill name>.24B <upgrada	ble>.B }*
```
Comment (aleos89) [inline src/map/clif.cpp:17236]: Minor spacing issue.

File: src/map/map-server.vcxproj [hunk 4]
```diff
@@ -368,5 +371,4 @@
     <Copy SourceFiles="$(SolutionDir)db\import-tmpl\statpoint.txt" DestinationFolder="$(SolutionDir)db\import\" ContinueOnError="true" Condition="!Exists('$(SolutionDir)db\import\statpoint.txt')" />
     <Copy SourceFiles="$(SolutionDir)db\import-tmpl\status_disabled.txt" DestinationFolder="$(SolutionDir)db\import\" ContinueOnError="true" Condition="!Exists('$(SolutionDir)db\import\status_disabled.txt')" />
   </Target>
-</Project>
-
+</Project>
```
Comment (aleos89) [inline src/map/map-server.vcxproj:374]: EOL space was removed.

File: src/map/map-server.vcxproj.filters [hunk 5]
```diff
@@ -269,4 +275,4 @@
       <Filter>Source Files</Filter>
     </ClCompile>
   </ItemGroup>
-</Project>
+</Project>
```
Comment (aleos89) [inline src/map/map-server.vcxproj.filters:278]: EOL space was removed.

**Comment by Masao87** [comment]
Not sure if this is a bug that only I have, but I don't seem to be able to set guild emblems.

screenLotusRO014
screenLotusRO015

**Comment by Haydrich** [comment]
Client side: 20200401
Issue: when using an item gives an Enable Skill ex: 28502 after take the item off the skill stuck in the skill bar 'but unable to use it'
1- @item 28502
2- wear it
3- Alt+s and check Other Skills
4- put the new skill in Skill Bar
5- take the item off
6- check if the Skill still in the bar or disappeared
image

image

**Comment by admkakaroto** [comment]
Skill bar are not auto refreshing.

**Comment by Razgrom** [comment]
noticed a problem with one of the skills RG - Earth Drive. After using it, there is a problem with the walk delay. if immediately after the skill, start moving, then his first steps do not go as usual, but the problem with animation. He makes at once in the region of 10-15 steps.It seems that this should not be so.

**Pull request merged**